File: QA-project/backend/src/db/schema/module-risk-scores.ts
import {
  pgTable,
  serial,
  integer,
  varchar,
  real,
  timestamp,
} from "drizzle-orm/pg-core";
import { relations } from "drizzle-orm";
import { datasets } from "./datasets";

export const moduleRiskScores = pgTable("module_risk_scores", {
  riskId: serial("risk_id").primaryKey(),
  datasetId: integer("dataset_id")
    .notNull()
    .references(() => datasets.datasetId, { onDelete: "cascade" }),
  moduleName: varchar("module_name", { length: 255 }).notNull(),
  bugCount: integer("bug_count").default(0),
  reopenRate: real("reopen_rate"),
  riskScore: real("risk_score"),
  computedAt: timestamp("computed_at").defaultNow().notNull(),
});

export const moduleRiskScoresRelations = relations(moduleRiskScores, ({ one }) => ({
  dataset: one(datasets, {
    fields: [moduleRiskScores.datasetId],
    references: [datasets.datasetId],
  }),
}));

File: QA-project/backend/app/models/module_risk_scores.py
from sqlalchemy import Column, Integer, String, Float, DateTime, ForeignKey, func
from sqlalchemy.orm import relationship
from app.database import Base


class ModuleRiskScore(Base):
    __tablename__ = "module_risk_scores"

    risk_id = Column(Integer, primary_key=True, autoincrement=True)
    dataset_id = Column(Integer, ForeignKey("datasets.dataset_id", ondelete="CASCADE"), nullable=False)
    module_name = Column(String(255), nullable=False)
    bug_count = Column(Integer, server_default="0")
    reopen_rate = Column(Float)
    risk_score = Column(Float)
    computed_at = Column(DateTime, server_default=func.now(), nullable=False)

    # Relationships
    dataset = relationship("Dataset", back_populates="module_risk_scores")
